qubes-rpc/qubes.StartApp

```diff
@@ -1,25 +1,50 @@
-#!/bin/sh
+#!/usr/bin/python3 --
 
-if [ -z "$1" ]; then
-    echo "This service requires an argument" >&2
-    exit 1
-fi
+import sys, os, pwd
+import qubesdb
+from qubesagent.xdg import launch
 
-app_basename="$1.desktop"
+def main(argv):
+    try:
+        (arg0, arg,) = argv
+    except ValueError:
+        print('This service requires an argument', file=sys.stderr)
+        sys.exit(1)
+    app_name = os.path.join('applications', arg + '.desktop')
+    try:
+        xdg_data_home = os.environ['XDG_DATA_HOME']
+    except KeyError:
+        xdg_data_home = os.path.join(pwd.getpwuid(os.getuid()).pw_dir, '.local/share')
+    try:
+        xdg_data_dirs = os.environ['XDG_DATA_DIRS']
+    except KeyError:
+        xdg_data_dirs = ['/usr/local/share','/usr/share']
+    else:
+        if '::' in xdg_data_dirs or \
+                xdg_data_dirs.startswith(':') or \
+                xdg_data_dirs.endswith(':'):
+            print('Empty entries in XDG_DATA_DIRS not allowed')
+            sys.exit(1)
+        xdg_data_dirs = xdg_data_dirs.split(':')
+    paths_to_try = [xdg_data_home] + xdg_data_dirs
+    qubes_db = qubesdb.QubesDB()
+    if qubes_db.read('/qubes-service/app-dispvm.' + arg) is not None:
+        try:
+            del os.environ['XDG_DATA_HOME']
```

Comment:
As discussed elsewhere, according to the spec, this will make an application use the default value. But there is also another concern - applications using `$XDG_DATA_HOME` for other things than just mime database. I don't see any obvious directories in _my_ `~/.local/share` that would be affected (be used by an application I'd like to use this feature on), but it is still a risk, and a reason to keep this (for now) as per-application feature.
In fact, similar risk applies to `$XDG_DATA_DIRS`, for example `/usr/share/icons`, or `/usr/share/fonts` (not sure if the latter is sensitive to `$XDG_DATA_DIRS`). For this, probably some symlinks will be enough.